src/ebay-textbox/textbox.tsx

```diff
@@ -56,34 +81,63 @@ const EbayTextbox: FC<EbayTextboxProps> = ({
         disabled: rest.disabled,
         label: floatingLabel,
         inputSize,
-        inputValue: controlledValue || value,
+        inputValue: controlledValue || inputValue,
         placeholder,
-        invalid
+        invalid,
+        onMount: onFloatingLabelInit
     })
 
     const handleFocus = (event?: FocusEvent<HTMLInputElement & HTMLTextAreaElement>) => {
-        onFocus(event)
+        onFocus(event, { value: event?.target?.value || defaultValue })
         onFloatingLabelFocus()
     }
 
     const handleBlur = (event: FocusEvent<HTMLInputElement & HTMLTextAreaElement>) => {
-        onBlur(event)
+        const newValue = event.target?.value
+        onBlur(event, { value: newValue })
+        if (newValue !== value) {
+            onChange(event, { value: newValue })
+            setValue(newValue)
+        }
         onFloatingLabelBlur()
     }
 
+    const handleKeyPress = (event?: KeyboardEvent<HTMLInputElement & HTMLTextAreaElement>) => {
+        const textbox = event.target as any
+        onKeyPress(event, { value: textbox?.value })
+    }
+    const handleKeyUp = (event?: KeyboardEvent<HTMLInputElement & HTMLTextAreaElement>) => {
+        const textbox = event.target as any
+        onKeyUp(event, { value: textbox?.value })
+    }
+    const handleKeyDown = (event?: KeyboardEvent<HTMLInputElement & HTMLTextAreaElement>) => {
+        const textbox = event.target as any
+        onKeyDown(event, { value: textbox?.value })
+    }
+
+    const handleInvalid = (event?: SyntheticEvent<HTMLInputElement & HTMLTextAreaElement>) => {
+        const textbox = event.target as any
+        onInvalid(event, { value: textbox?.value })
+    }
```

Comment:
I don't think we need to pass the `{ value }` here, they have access that in the event.target, we should try to be as much as possible near to the native API.

Reply:
what do you mean by native API? I'm just following ebayUI API:
<img width="502" alt="image" src="https://github.com/eBay/ebayui-core-react/assets/2737310/42ecc671-6c9f-42a6-86d0-cee0cf344687">

Reply:
I mean the HTML/JS/DOM api. We can access it with `event.target.value`